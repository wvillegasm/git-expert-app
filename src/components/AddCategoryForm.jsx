import { useState } from "react";

export const AddCategoryForm = ({ onAddCategory }) => {
  const [gifCategory, setGifCategory] = useState("");

  const handleOnChange = (e) => {
    setGifCategory(e.target.value);
  };

  const handleOnSubmit = (e) => {
    e.preventDefault();

    const gifCategoryFormatted = gifCategory.trim().toUpperCase();

    if (gifCategoryFormatted.length < 2) {
      return;
    }

    onAddCategory(gifCategoryFormatted, setGifCategory);

    // setGifCategory("");
  };

  return (
    <form className="space-bottom" onSubmit={handleOnSubmit}>
      <input
        type="text"
        placeholder="Search gifs"
        onChange={handleOnChange}
        value={gifCategory}
      />
    </form>
  );
};
