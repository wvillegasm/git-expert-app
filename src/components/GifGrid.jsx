import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './';

import '../styles.css';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {isLoading && <p>Loading...</p>}

      <div className='gif-grid'>
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
