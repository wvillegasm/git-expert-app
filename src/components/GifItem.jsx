export const GifItem = ({ title, url }) => {
  return (
    <div className='gif-item'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
