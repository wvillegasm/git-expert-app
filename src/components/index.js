export * from './AddCategoryForm';
export * from './GifGrid';
export * from './GifItem';
