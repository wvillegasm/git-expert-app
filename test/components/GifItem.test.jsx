import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { GifItem } from "../../src/components/GifItem";

describe("GifItem Component", () => {
  const title = "Dragon Ball";
  const url = "https://example.com/goku.gif";

  test("should match snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should render image with correct URL and ALT attributes", () => {
    render(<GifItem title={title} url={url} />);

    const img = screen.getByRole("img");
    expect(img.src).toBe(url);
    expect(img.alt).toBe(title);
  });

  test("should render title in a paragraph", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  test("should have gif-item class", () => {
    render(<GifItem title={title} url={url} />);

    const div = screen.getByRole("img").parentElement;
    expect(div.className).toContain("gif-item");
  });
});
