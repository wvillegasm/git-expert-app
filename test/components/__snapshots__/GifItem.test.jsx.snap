// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifItem Component > should match snapshot 1`] = `
<div>
  <div
    class="gif-item"
  >
    <img
      alt="Dragon Ball"
      src="https://example.com/goku.gif"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
